astropy/coordinates/tests/test_intermediate_transformations.py

```diff
@@ -519,11 +519,12 @@ def test_teme_itrf():
 
 
 @pytest.mark.remote_data
-def test_earth_orientation_table():
+def test_earth_orientation_table(monkeypatch):
     """Check that we can set the IERS table used as Earth Reference.
 
     Use the here and now to be sure we get a difference.
     """
+    monkeypatch.setattr('astropy.utils.iers.conf.auto_download', True)
```

Comment:
Usually not a fan of monkeypatching but it does seem less clunky than the `try ... finally ...` implementation I did in #10311, so 👍

Reply:
There is also `with iers.iers.Conf.auto_download.set_temp(True)` but I felt this was a bit more natural. I actually rather liked your `setup_class` and `teardown_class`, but it seemed too much effort for a single function.